app/services/course/assessment/answer/auto_grading_service.rb

```diff
@@ -28,12 +40,11 @@ def pick_grader(question)
   # Subclasses should call this implementation after they are done to persist the changes to the
   # database.
   #
-  # @param [Course::Assessment::Answer::AutoGrading] auto_grading The object to store grading
-  #   results in.
-  # @return [Boolean] True if the grading could be saved.
-  def grade(auto_grading)
-    auto_grading.answer.publish!
-    auto_grading.answer.grader = User.system
-    auto_grading.save
+  # @param [Course::Assessment::Answer] answer The answer to be graded.
+  # @return [Course::Assessment::Answer] The graded answer, not that the answer is not saved.
+  def grade(answer)
+    answer.publish!
+    answer.grader = User.system
+    answer
```

Comment:
For the current use case this seems to make sense.

I'm trying to figure out the pros/cons between using `answer` vs `auto_grading` in the long term - it seems like the original `auto_grading` not only is the backbone for autograding service, but also provides more abstraction for future extension (although I'm not sure what else it could be, autograde a few answers at 1 go, or something else?). Any thoughts?